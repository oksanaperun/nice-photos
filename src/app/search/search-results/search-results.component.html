<app-search-results-total-count [totalCount]="data.totalCount"></app-search-results-total-count>
<app-search-tips *ngIf="showTips; else itemList"></app-search-tips>
<ng-template #itemList>
  <app-search-results-item-list [items]="data.items"></app-search-results-item-list>
</ng-template>
