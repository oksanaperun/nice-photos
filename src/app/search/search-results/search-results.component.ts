import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
import { SearchResultsData } from '../search';

@Component({
  selector: 'app-search-results',
  templateUrl: './search-results.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})

export class SearchResultsComponent {
  @Input() data: SearchResultsData;

  get showTips(): boolean {
    return this.data.totalCount === 0;
  }
}
