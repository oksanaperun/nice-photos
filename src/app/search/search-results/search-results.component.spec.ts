import { TestBed, ComponentFixture } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { By } from '@angular/platform-browser';
import { SearchResultsComponent } from './search-results.component';

describe('SearchResultsComponent', () => {
  let component: SearchResultsComponent;
  let fixture: ComponentFixture<SearchResultsComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [SearchResultsComponent],
      schemas: [NO_ERRORS_SCHEMA]
    });

    fixture = TestBed.createComponent(SearchResultsComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
  });
});
