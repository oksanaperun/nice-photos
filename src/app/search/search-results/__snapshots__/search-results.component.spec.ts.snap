// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsComponent should render item list when total items count is greater than zero 1`] = `
<app-search-results
  data={[Function Object]}
>
  <app-search-results-total-count />
  <app-search-results-item-list />
  
  
</app-search-results>
`;

exports[`SearchResultsComponent should render search tips when total items count equals zero 1`] = `
<app-search-results
  data={[Function Object]}
>
  <app-search-results-total-count />
  
    <app-search-tips />
  
  
  
</app-search-results>
`;
